File: Java/program64.c
///////////////////////////////////////////////////////
//
//  File name :     program64.c
//  Descreption :   Accept Input and Calculate Summation of Factors as well as Non-Factors of Number and Display
//  Author :        Vaibhav
//  Date :          20/05/2025
//
///////////////////////////////////////////////////////

#include<stdio.h>

void DisplaySumFactorsNonFactors(int iValue)
{
    int iCnt = 0;
    int iSumFact = 0,iSumNonFact = 0;

    for(iCnt = 1;iCnt <= iValue; iCnt++)
    {
        if((iValue % iCnt) == 0)
        {
            iSumFact = iSumFact + iCnt;
        }
        else
        {
            iSumNonFact = iSumNonFact + iCnt;
        }
    }
    printf("Summation of Factors are : %d\n",iSumFact);
    
    printf("Summation of Non-Factors are : %d\n",iSumNonFact);

}

int main()
{
    int iNo = 0;
    int iRet = 0;

    printf("---- This Application is used to used to Display Summation of Factors as well as Non-Factors of Number ----\n");

    printf("Enter Number: ");
    scanf("%d",&iNo);

    DisplaySumFactorsNonFactors(iNo);
    
    return 0;
}

File: program64.java
///////////////////////////////////////////////////////
//
//  File name :     program64.java
//  Descreption :   Accept Input and Calculate Summation of Factors as well as Non-Factors of Number and Display
//  Author :        Vaibhav
//  Date :          
//
///////////////////////////////////////////////////////

import java.util.*;

class Number
{
    public void Sum(int iNo)
    {
        int iCnt = 0;
        int iSum1 = 0,iSum2 = 0;
        for(iCnt = 1;iCnt <= iNo;iCnt++)
        {
            if((iNo % iCnt) != 0)       //dry run
            {
                iSum1 = iSum1 + iCnt;
            }
            else
            {
                iSum2 = iSum2 + iCnt;
            }
        }
        System.out.println("Summation of Factors are : "+iSum2);            //python can return 2 values also
        System.out.println("Summation of Non-Factors are : "+iSum1);
    }
}

class program64
{
    public static void main(String A[])
    {
        Scanner sobj = new Scanner(System.in);
        
        System.out.println("---- This Application is used to Display Summation of Factors as well as Non-Factors of Number ----");

        System.out.println("Enter number: ");
        int iValue = sobj.nextInt();

        Number nobj = new Number();
        nobj.Sum(iValue) ;

        
    }
}